### PR TITLE
refactor(torii): update query logic for them to not block

### PR DESCRIPTION
**Pull request by lambda-0x**
## Summary by CodeRabbit

- **New Features**
	- Enhanced query handling with the introduction of a `QueryType` for better categorization during execution.
	- Streamlined SQL operations for inserting and updating entities, improving efficiency and clarity.
	- Refactored database interaction into a dedicated asynchronous function for improved modularity.

- **Bug Fixes**
	- Removed redundant code to reduce complexity and potential performance issues.

- **Refactor**
	- Improved code organization by encapsulating database logic and enhancing separation of concerns.

### CODE REVIEW

**Review by glihm — APPROVED** [review]
Looks good to have write and read separated in preparation for multithreading and also allows the batching of inserts.

@Larkooo any additional comment on this one?

File: crates/torii/core/src/query_queue.rs [hunk 1]
```diff
@@ -69,20 +91,54 @@ impl QueryQueue {
...
+                query = match arg {
+                    Argument::Null => query.bind(None::<String>),
+                    Argument::Int(integer) => query.bind(integer),
+                    Argument::Bool(bool) => query.bind(bool),
+                    Argument::String(string) => query.bind(string),
+                    Argument::FieldElement(felt) => query.bind(format!("{:#x}", felt)),
+                }
             }
+
+            let broker_message = match query_type {
```
Comment (Larkooo) [inline crates/torii/core/src/query_queue.rs:119]: Does this mean we also fetch the entity for each set entity we do, to get the new EntityUpdated? Let's either make the queries `RETURNING *` and use that or just initialize the EntityUpdated without fetching it

Comment (glihm) [reply crates/torii/core/src/query_queue.rs:119]: @lambda-0x any update on this based on the @Larkooo comment?

Comment (lambda-0x) [reply crates/torii/core/src/query_queue.rs:119]: oh my bad, i made the changes yesterday but forgot to push

**Pull request merged**